app/hooks/StrapiAPIHooks.ts

```diff
@@ -0,0 +1,31 @@
+import useSWR from "swr"
+import fetcher from "utils/fetcher";
+import config from "../config";
+import { type Footer } from "models/Strapi";
+
+interface SWR {
+  error?: Error;
+  isLoading: boolean;
+}
+
+interface FooterData extends SWR {
+  data: Footer;
+}
+
+export function useFooterData(): FooterData {
```

Comment:
the idea is that we would create a hook for each API call we need to make

Reply:
I like it. How about adding another light layer of abstraction for an endpoint builder. Something like `createStrapiHook`. Then you might export an interface object that looks like:

```typescript
const BASE_PATH = '/api`;

// needs types, of course
function createStrapiHook(path, requestParams)  {
	const { data, error, isLoading } = useSWR(
    path,
    fetcher.bind({},
      `${config.STRAPI_API_URL}/${BASE_PATH}/footer?populate[links][populate]=*`,
      {
        'Authorization': `Bearer ${config.STRAPI_API_TOKEN}`
      },
      ...requestParams, // <-- additional params passthrough or similar
    )
  );

  return {
    data: data?.data?.attributes,
    error,
    isLoading
  };
}

export default {
	useFooterData: createStrapiHook({ url: '/footer'}),
	useNavigationData: createStrapiHook({ url: '/navigation'}),
	// ...etc
}
```

Reply:
I went back and forth on this if we should do it now or. wait until we get a couple more in there, but they all should be generally the same so went ahead with your suggestion. Let me know what you think.

Reply:
Looks great.